posts/advancing from micro services to including micro frontends and how this leads to vertical teams.md

```diff
@@ -0,0 +1,101 @@
+
+![Monolith Architecture Diagram](https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2022/12/monolith.png) 
+- ***4 components bundled within 1 application***
+
+### Some Monolithic Architecture Disadvantages
+
+Teams working in this environment risk affecting the work of other teams in the application. Since developers may have different levels of experience, backgrounds, and coding styles, this could increase the chances of bugs or inefficient code within the application. The development process of this type of application has less flexibility, more tangled parts, and a tightly-coupled type of infrastructure. The application components are developed, scaled, and deployed as one unit; therefore, every change made would require the deployment of the whole application. Imagine having to deploy the entire application multiple times a day every time there is a need to make changes to a single component. The release process of an application will be much longer, potentially causing downtime and reducing productivity. This is also the case with scaling a small functionality within a single component of an application.  
+
+Once upon a time, the monolithic approach to development may have been the standard way of composing applications. However, as mentioned above, companies needed a more efficient way to develop, scale, and deploy their applications.
+
+## Microservices
+
+### What are Microservices?
+
+[Microservices](https://aws.amazon.com/microservices) are small individual components completely independent of other components of the application. Essentially, they are loosely-coupled, self-contained services that makeup one whole extensive application. Following the [Single-Responsibility Principle](https://en.wikipedia.org/wiki/Single-responsibility_principle) of [SOLID](https://en.wikipedia.org/wiki/SOLID), microservices are mainly made up of one service, one process, one job, or one function. In essence, each microservice should have one single domain. Therefore, changes to a microservice prevent affecting other components of the application due to its natural encapsulation. Unlike the monolithic architecture, microservices are developed, scaled, and deployed separately. However, they are still part of the application and can communicate with one another via APIs, Message Brokers, or Service Mesh.    
+ 
+Microservices also eliminate the issue of deploying the entire application for a single component or feature. Microservices allow the deployment of a single component with new changes, increasing the release process timeline. Microservices also enable teams to work independently on a single component without affecting other teams working on different application sections. This workflow gives developers more flexibility to code in the language or framework of their choice. 
```

Comment:
I feel like the microservice section is talking about a MFE instead of an API.

Reply:
made the suggested changes